### ISSUE TITLE
Aqara Gateway in HA geys "Aborted"

### ISSUE THREAD

**Issue by HazardousRat** [niceboygithub/AqaraGateway#83]
I have an Aqara M1S, CN (ZHWG15LM) and have successfully enabled telnet, flashed firmware per the guide, added the AqaraGateway in HACS and every time I try to submit I get an aborted message. Can someone please elaborate on how to successfully add the M1S to HA while also having/using the device in Aqara home?  The M1S is still in Mi Home mode. I tried to switch to Aqara Home, it sees the device but fails to successfully connect.

**Comment by niceboygithub** [comment]
Which version of HA?

**Comment by HazardousRat** [comment]
HA version 2022.2.0

**Comment by niceboygithub** [comment]
> I received this error in my logs.
> 
> Source: util/async_.py:144
> 
> Detected blocking call inside the event loop. This is causing stability issues. Please report issue to the custom component author for aqara_gateway doing blocking calls at custom_components/aqara_gateway/core/shell.py, line 79: time.sleep(.1)
> 
> I'm not sure if this is the reason why trying to add the AqaraGateway was aborted or if this is one of several. Any help in resolving this would be greatly appreciated.

This is just warning message.

You may need to check are your gateway and HA in the same domain.

**Comment by HazardousRat** [comment]
No, I have the gateway and HA on two separate networks one being (the gateway) on a vlan. Does this integration require both devices to be on the same subnet?

**Comment by HazardousRat** [comment]
I have the gateway on an "IoT" vlan via pfsense  that I like to keep separate from my HA. If this requires both to be on the same subnet then I'm not sure why was I able to successfully telnet and place firmware on the device while both were on two separate networks.

**Comment by HazardousRat** [comment]
Also when trying to add the Aqara Gateway M1S, CN (ZHWG15LM) in to HA should the device remain in MI Home Mode or Aqara Home mode? If you or anyone have had success in migrating the Gateway to HA please explain because I keep getting the aborted message. Just for "testing" purposes I moved both HA & the Gateway onto the same subnet and getting the same "Abort" message in HA.

**Comment by bluefoxlee** [comment]
HA Supervised 2022.2.3
Got same error when try to add E1 CN Gateway.

**Comment by niceboygithub** [comment]
> HA Supervised 2022.2.3
> Got same error when try to add E1 CN Gateway.

HA 2022.2.3 had some changes make this error. I will debug it

**Comment by bluefoxlee** [comment]
> HA 2022.2.3 had some changes make this error. I will debug it

Got it. Thanks.
